scripts/build_central_heating_temperature_profiles/run.py

```diff
@@ -130,45 +130,129 @@ def map_temperature_dict_to_onshore_regions(
     )
 
 
+def scale_temperature_to_investment_year(
+    temperature_baseyear: dict,
+    relative_annual_temperature_reduction: float,
+    investment_year: int,
+    current_year: int,
+) -> dict:
+    """
+    Scale temperature for each country to investment year by constant
+    exponential factor.
+
+    Parameters
+    ----------
+    temperature_baseyear : dict
+        Dictionary with the current temperatures as values and country keys as keys.
+    relative_annual_temperature_reduction : float
+        The annual reduction rate of the temperature, must be between 0 and 1.
+    investment_year : int
+        The year in which the investment is planned.
+    current_year : int
+        The current year.
+
+    Returns
+    -------
+    dict
+        A dictionary with the temperature for each country in the investment year.
+
+    Raises
+    ------
+    ValueError
+        If the investment year is before the current year.
+        If the relative annual temperature reduction is not between 0 and 1.
+    """
+
+    if investment_year < current_year:
+        raise ValueError(
+            f"Error: Investment year {investment_year} is before current year {current_year}."
+        )
+    if (
+        relative_annual_temperature_reduction < 0
+        or relative_annual_temperature_reduction > 1
+    ):
+        raise ValueError(
+            f"Error: Relative annual temperature reduction must be between 0 and 1, but is {relative_annual_temperature_reduction}."
+        )
+
+    return {
+        key: temperature_baseyear[key]
+        * (
+            (1 - relative_annual_temperature_reduction)
+            ** (investment_year - current_year)
+        )
+        for key in temperature_baseyear.keys()
+    }
+
+
 if __name__ == "__main__":
     if "snakemake" not in globals():
         from _helpers import mock_snakemake
 
         snakemake = mock_snakemake(
             "build_cop_profiles",
             clusters=48,
+            planning_horizons="2050",
         )
 
     set_scenario_config(snakemake)
 
-    max_forward_temperature = snakemake.params.max_forward_temperature_central_heating
-    min_forward_temperature = extrapolate_missing_supply_temperatures_by_country(
-        extrapolate_from=max_forward_temperature,
-        extrapolate_to=snakemake.params.min_forward_temperature_central_heating,
+    # investment_year = int(snakemake.wildcards.planning_horizons[-4:])
+    # reduce temperatures to investment years by constant exponential factor
+    max_forward_temperature_investment_year = scale_temperature_to_investment_year(
+        temperature_baseyear=snakemake.params.max_forward_temperature_central_heating_baseyear,
+        relative_annual_temperature_reduction=snakemake.params.relative_annual_temperature_reduction,
+        investment_year=int(snakemake.wildcards.planning_horizons[-4:]),
```

Comment:
```suggestion
        investment_year=int(snakemake.wildcards.planning_horizons),
```
I removed the unnecessary `[-4:]` recently.